src/hangar/dataloaders/common.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+from typing import Iterable, Optional, Union
+
+import numpy as np
+
+DatasetsRef = Union['DatasetDataReader', Iterable['DatasetDataReader']]
+
+
+class GroupedDsets(object):
+    """Groups hangar datasets and validate suitability for usage in dataloaders.
+
+    It can choose a subset of samples in the hangar datasets by checking the
+    list of keys or an index range. :class:`GroupedDsets` expect all the input
+    hangar datasets to have same length and same keys. Although it checks for
+    the length of all the datasets, it doesn't do any explicit check for the
+    sample names and hence the dataloaders while fetching the data will throw
+    `KeyError` in case of non-similar keys.
+    """
+
+    def __init__(self,
+                 hangar_datasets: DatasetsRef,
+                 keys: Optional[Iterable[Union[int, str]]] = None,
+                 index_range: Optional[slice] = None):
+
+        if not isinstance(hangar_datasets, (list, tuple)):
+            hangar_datasets = (hangar_datasets,)
+
+        dset_keys, dset_lens = {}, []
+        for dataset in hangar_datasets:
+            kset = set(dataset.keys())
+            dset_keys[dataset.name] = kset
+            dset_lens.append(len(kset))
+
+        if not np.allclose(dset_lens):
+            warnings.warn('Datasets do not contain equal num samples', UserWarning)
+
+        if (keys is not None) and (not isinstance(keys, (list, tuple, set))):
```

Comment:
What is this doing? It's not clear what wrapping `keys` in a tuple is supposed to do.

Reply:
Allowing users to pass single key (A string)